### PR TITLE
Add modifier to overwrite test configuration by flags

### PR DESCRIPTION
**Pull request by jprzychodzen**
Flag level allowing configuration of config for ClusterLoader2

It solves problem for #1282 and little more.

It does not support slices on parameter path, but this feature is currently not needed.

/assign @mm4tt 
/cc @wojtek-t

### CODE REVIEW

**Comment by mm4tt** [comment]
Nice!

/lgtm
/approve

File: clusterloader2/pkg/modifier/modifier.go [hunk 1]
```diff
@@ -60,3 +75,69 @@ func (m *simpleModifier) ChangeTest(c *api.Config) {
 		}
 	}
 }
+
+func (m *simpleModifier) modifyOverwrite(c *api.Config) error {
+	for _, o := range m.overwriteTestConfig {
+		kv := strings.Split(o, "=")
+		if len(kv) != 2 {
+			errMsg := fmt.Sprintf("Not a key=value pair: %s", o)
+			klog.Error(errMsg)
```
Comment (wojtek-t) [inline clusterloader2/pkg/modifier/modifier.go:84]: don't log errors if you return them - it's responsibility of the calls to do whether they want with it

Comment (wojtek-t) [reply clusterloader2/pkg/modifier/modifier.go:84]: Same everywhere below

File: clusterloader2/pkg/modifier/modifier.go [hunk 2]
```diff
@@ -60,3 +75,69 @@ func (m *simpleModifier) ChangeTest(c *api.Config) {
...
+				// If path came across ptr to nil, we need to create zero value before dereferencing
+				if curValue.IsNil() {
+					expectedType := curValue.Type().Elem()
+					pointerToZeroValue := reflect.New(expectedType)
+					curValue.Set(pointerToZeroValue)
+				}
+				curValue = curValue.Elem()
+			}
+		}
+		e := m.overwriteValue(curValue, v, k)
```
Comment (wojtek-t) [inline clusterloader2/pkg/modifier/modifier.go:107]: using "e" as error is a bit misleading...

File: clusterloader2/pkg/modifier/modifier.go [hunk 3]
```diff
@@ -60,3 +75,69 @@ func (m *simpleModifier) ChangeTest(c *api.Config) {
...
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		intV, err := strconv.ParseInt(v, 10, 64)
+		if err != nil {
+			errMsg := fmt.Sprintf("test config overwrite error: Cannot parse '%s' for key '%s' to int: %v", v, k, err)
+			klog.Error(errMsg)
+			return errors.New(errMsg)
+		}
+		klog.Infof("Setting int value '%d' for key '%s'", intV, k)
+	default:
+		errMsg := fmt.Sprintf("Unsupported kind: %v for key %s", node.Kind(), k)
```
Comment (wojtek-t) [inline clusterloader2/pkg/modifier/modifier.go:138]: errors should start with small letter

File: clusterloader2/pkg/modifier/modifier.go [hunk 4]
```diff
@@ -60,3 +75,69 @@ func (m *simpleModifier) ChangeTest(c *api.Config) {
...
+		klog.Infof("Setting string value '%s' for key '%s'", v, k)
+		node.SetString(v)
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		intV, err := strconv.ParseInt(v, 10, 64)
+		if err != nil {
+			errMsg := fmt.Sprintf("test config overwrite error: Cannot parse '%s' for key '%s' to int: %v", v, k, err)
+			klog.Error(errMsg)
+			return errors.New(errMsg)
+		}
+		klog.Infof("Setting int value '%d' for key '%s'", intV, k)
```
Comment (wojtek-t) [inline clusterloader2/pkg/modifier/modifier.go:136]: Seems that you don't really set that :)

File: clusterloader2/pkg/modifier/modifier.go [hunk 5]
```diff
@@ -17,41 +17,56 @@ limitations under the License.
...
+	skipSteps           []string
 }
 
 // Ensuring that simpleModifier implements Modifier interface
 var _ Modifier = &simpleModifier{}
 
-func (m *simpleModifier) ChangeTest(c *api.Config) {
+func (m *simpleModifier) ChangeTest(c *api.Config) error {
+	m.modifySkipSteps(c)
+	err := m.modifyOverwrite(c)
```
Comment (wojtek-t) [inline clusterloader2/pkg/modifier/modifier.go:57]: nit: return m.modifyOverwrite(c)

**Pull request merged**